skill-tree/src/test/java/com/RDS/skilltree/integration/EndorsementsIntegrationTests.java

```diff
@@ -0,0 +1,442 @@
+
+        Response response =
+                given()
+                        .cookies(RestAPIHelper.getGuestUserCookie())
+                        .get("/v1/endorsements?skillID=" + skillID + "&userID=" + userID);
+
+        response
+                .then()
+                .statusCode(200)
+                .contentType("application/json")
+                .body("content", everyItem(hasKey("skill_id")))
+                .body("content.skill_id", everyItem(equalTo(skillID)))
+                .body("content", everyItem(hasKey("user_id")))
+                .body("content.user_id", everyItem(equalTo(userID)));
+    }
+
+    @Test
+    @Disabled
+    @DisplayName(
+            "Return 204, when there are no endorsements present for the given userID in UUID form")
+    public void itShouldReturn204OnEndorsementSearchWithValidUserIDButNotPresentInList() {
+        String userID = UUID.randomUUID().toString();
+
+        Response response =
+                given()
+                        .cookies(RestAPIHelper.getGuestUserCookie())
+                        .get("/v1/endorsements?userID=" + userID);
+
+        response.then().statusCode(204);
+    }
+
+    @Test
+    @Disabled
+    @DisplayName(
+            "Return 204, when there are no endorsements present for the given skillID in UUID form")
+    public void itShouldReturn204OnEndorsementSearchWithValidSkillIDButNotPresentInList() {
+        String skillID = UUID.randomUUID().toString();
+
+        Response response =
+                given()
+                        .cookies(RestAPIHelper.getGuestUserCookie())
+                        .get("/v1/endorsements?skillID=" + skillID);
+
+        response.then().statusCode(204);
+    }
+
+    @Test
+    @Disabled
+    @DisplayName("Return 400, given a userID which is not a UUID")
+    public void itShouldReturn400OnEndorsementSearchWithInvalidUserID() {
+        String userID = "invalid-user-id";
+
+        Response response =
+                given()
+                        .cookies(RestAPIHelper.getGuestUserCookie())
+                        .get("/v1/endorsements?userID=" + userID);
+
+        response.then().statusCode(400);
+    }
+
+    @Test
+    @Disabled
+    @DisplayName("Return 400, given a skillID which is not a UUID")
+    public void itShouldReturn400OnEndorsementSearchWithInvalidSkillID() {
+        String skillID = "invalid-skill-id";
+
+        Response response =
+                given()
+                        .cookies(RestAPIHelper.getGuestUserCookie())
+                        .get("/v1/endorsements?skillID=" + skillID);
+
+        response.then().statusCode(400);
+    }
+
+    @Test
+    @Disabled
+    @DisplayName("Return 204, given an offset value greater than maximum endorsements")
+    public void itShouldReturn204OnEndorsementSearchWithOffsetGreaterThanMaximumEndorsements() {
+        Response response =
+                given().cookies(RestAPIHelper.getGuestUserCookie()).get("/v1/endorsements?offset=100");
```

Comment:
why do we need to check with offset...?

Reply:
I observed this edge case of out-of-bound and negative page offsets, testing to make sure we return an empty page with 204 and not throw an error. This is related to pagination and not exactly the search functionality, should I remove these tests?